File: src/nomanssave/cw.java
package nomanssave;

import java.io.File;
import javax.swing.Icon;
import javax.swing.filechooser.FileView;

class cw extends FileView {
   // $FF: synthetic field
   final cv fR;

   cw(cv var1) {
      this.fR = var1;
   }

   public Icon getIcon(File var1) {
      String var2 = var1.getName();
      return (Icon)(var2.endsWith(".wp0") ? cv.as() : super.getIcon(var1));
   }

   public String getName(File var1) {
      String var2 = var1.getName();
      return var2.endsWith(".wp0") ? var2.substring(0, var2.length() - 4) : var2;
   }
}

File: NMSSaveEditor/nomanssave/lower/cw.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

namespace NMSSaveEditor
{



public class cw : FileView {
   public string Name => getName();
   public cv fR;

   public cw(cv var1) {
      this.fR = var1;
   }

   public Icon getIcon(FileInfo var1) {
      string var2 = var1.Name;
      return (Icon)(var2.EndsWith(".wp0") ? cv.@as() : base.getIcon(var1));
   }

   public string getName(FileInfo var1) {
      string var2 = var1.Name;
      return var2.EndsWith(".wp0") ? var2.Substring(0, var2.Length - 4) : var2;
   }
}



}
